ably/rest_client.go

```diff
@@ -27,36 +29,30 @@ func query(fn func(string, interface{}) (*http.Response, error)) QueryFunc {
 }
 
 type RestClient struct {
-	Auth *Auth
-
-	RestEndpoint string
-	Protocol     string
+	Options *ClientOptions
+	Auth    *Auth
 
-	HTTPClient *http.Client
-
-	channels map[string]*RestChannel
-	chanMtx  sync.Mutex
+	mtx   sync.Mutex // protects chans
```

Comment:
That looks weird. Renaming a variable which obfuscates its purpose and forcing us to do a look up to see that it `// protects chans`?